http-clients/src/main/java/org/triplea/http/client/lobby/chat/LobbyChatClient.java

```diff
@@ -41,37 +41,37 @@ public static LobbyChatClient newClient(
   }
 
   public void setChatMessageListeners(final ChatMessageListeners chatMessageListeners) {
-    setListeners(chatMessageListeners);
+    new WebsocketListenerBinding<>(webSocketClient, chatMessageListeners) {
+      @Override
+      protected ChatServerMessageType readMessageType(
+          final ServerMessageEnvelope serverMessageEnvelope) {
+        return ChatServerMessageType.valueOf(serverMessageEnvelope.getMessageType());
+      }
+    };
```

Comment:
If you don't know that the constructor has side-effects and registers itself as part of the constructor, this line looks very weird.
I think you should be able to move the registration thingy to a seperate method to make this more explicit

Reply:
The constructor having side-effects becomes obvious unless someone believes this is dead-code. Having such a construct is very odd though, I would agree.  

I think this would get fixed soon: https://github.com/triplea-game/triplea/issues/5845; fixing it now would more likely result in code-churn and would be quite piece-meal.